### PR TITLE
fix(ui): add missing cy.testA11y() coverage to Cypress page objects and tests

### PR DESCRIPTION
**Pull request by manaswinidas**
## Description

Add accessibility tests in all Cypress tests. This is just test change - not frontend changes.

## How Has This Been Tested?

All existing tests should pass

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [x] The commits have meaningful messages
- [x] Automated tests are provided as part of the PR for major new functionalities; testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).
- [ ] **For first time contributors**: Please reach out to the [Reviewers](https://github.com/kubeflow/hub/blob/main/OWNERS) to ensure all tests are being run, ensuring the label `ok-to-test` has been added to the PR.

If you have UI changes

- [x] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/frontend/src/app/pages/modelRegistry/components/pf-overrides/WarningAlertA11y.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.mr-warning-alert--accessible-title {
```
Comment (manaswinidas) [inline clients/ui/frontend/src/app/pages/modelRegistry/components/pf-overrides/WarningAlertA11y.scss]: This is intentional and fixes accessibility violations(Cypress tests were failing otherwise)- I've reached out to the PF team to file a bug - once I have the link - I'll attach it here

Comment (manaswinidas) [reply clients/ui/frontend/src/app/pages/modelRegistry/components/pf-overrides/WarningAlertA11y.scss]: removing and checking whether I still see the violation

File: clients/ui/frontend/src/__tests__/cypress/cypress/tests/mocked/modelRegistry/modelVersionsCard.cy.ts [hunk 2]
```diff
@@ -284,6 +284,8 @@ describe('Model Versions Card', () => {
         items: [],
       }),
     );
+    cy.visit('/model-registry/modelregistry-sample/registered-models/1/overview');
+    appChrome.waitForA11y();
```
Comment (ppadti) [inline clients/ui/frontend/src/__tests__/cypress/cypress/tests/mocked/modelRegistry/modelVersionsCard.cy.ts:288]: Why are we just adding `appChrome.waitForA11y();` to just this test and the rest uses cy.visit() without a11y

File: clients/ui/frontend/src/__tests__/cypress/cypress/tests/mocked/modelRegistry/modelDetailsCard.cy.ts [hunk 3]
```diff
@@ -105,6 +106,7 @@ describe('Model Details Card', () => {
 
   it('displays model details correctly', () => {
     cy.visit('/model-registry/modelregistry-sample/registered-models/1/overview');
+    appChrome.waitForA11y();
```
Comment (ppadti) [inline clients/ui/frontend/src/__tests__/cypress/cypress/tests/mocked/modelRegistry/modelDetailsCard.cy.ts:109]: same here as https://github.com/kubeflow/hub/pull/2793/changes#r3372059170

**Review by ppadti — CHANGES_REQUESTED** [review]
Is there any reason for adding cy.testA11y() / appChrome.waitForA11y() to only some tests and not all?
These tests still navigates without a11y check:(few tests have and not all)
modelVersionsCard.cy.ts
modelDetailsCard.cy.ts
mcpServerDetails.cy.ts
modelRegistry.cy.ts
modelCatalogDetails.cy.ts
modelCatalogTabs.cy.ts
modelCatalogCard.cy.ts

File: clients/ui/frontend/src/__tests__/cypress/cypress/tests/mocked/modelCatalog/modelCatalogDetails.cy.ts [hunk 4]
```diff
@@ -251,6 +260,8 @@ describe('Model Catalog Details Page - Edge Cases', () => {
     modelCatalog.visit();
     modelCatalog.findLoadingState().should('not.exist');
     modelCatalog.findModelCatalogDetailLink().first().click();
+    // Skip appChrome.waitForA11y() - blocked by focus trap a11y violation
+    // on the disabled Register model button (Popover → Tooltip fix in #2817).
```
Comment (manaswinidas) [inline clients/ui/frontend/src/__tests__/cypress/cypress/tests/mocked/modelCatalog/modelCatalogDetails.cy.ts:264]: we should enable this in #2817 else here if that goes in first to double check

**Comment by ppadti** [comment]
This branch has merge conflict - could you please resolve this

**Review by ppadti — CHANGES_REQUESTED** [review]
One minor comment is that - I am still not seeing the cold start data on the cards - because we have this `findMatchingHardwareConfig` which matches the cold start with hardware configuration data and in the mock we added here does not have matching hardware config for cold start

**Review by ppadti — APPROVED** [review]
/lgtm
/approve
